### ISSUE TITLE
discrepancy b/w val_loss and val_mse

### ISSUE THREAD

**Issue by 9rince** [keras-team/keras#13881]
I am passing in all the many error functions as metric values in Keras while compiling the model as follows, 
```
model.compile(loss='mean_squared_error' , optimizer='sgd', metrics=['mean_squared_error',
                                                                    'mean_absolute_error',
                                                                    'mean_absolute_percentage_error',
                                                                    'mean_squared_logarithmic_error',
                                                                    'squared_hinge',
                                                                    'hinge',
                                                                    'logcosh',
                                                                    'kullback_leibler_divergence',
                                                                    'poisson',
                                                                    'cosine_proximity'])
```
but I see the value of val_loss and val_mean_squared_error different from each other. They should be same right? what am I missing ?

**Comment by nperraud** [comment]
I observed the same thing. According to me this is probably a bug...

**Comment by sjwetzel** [comment]
Observed the same thing while training on generators.

**Comment by thiagodelatorre** [comment]
This issue happened to me when I changed my generator in a way that it started changing its state ("side-effect"). Removing any side-effects from the generator fixed the issue.

**Comment by jvishnuvardhan** [comment]
@9rince This is a stale issue.

May be this was an issue with older versions. But, I don't see any discrepancy with `TF2.6`. Please note that i tried with a simple model and noticed `loss` and `metric` are matching when using `mean_squared_error`. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/7853d34204e9cec1f288bf764b2a561a/tensorflow_tensorboard_regression.ipynb).

Please verify once and close the issue if this was resolved for you when using recent TF/keras versions. Thanks!

**Comment by jvishnuvardhan** [comment]
Closing the issue as this was resolved. Please feel free to reopen if the issue persists for you. Thanks!
